File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/api_service/firebase/FireBaseApiService.java
package sabalan.paydar.mohtasham.ezibazi.api_service.firebase;

import android.content.Context;


import org.json.JSONException;
import org.json.JSONObject;


import sabalan.paydar.mohtasham.ezibazi.api_service.ApiRequest;
import sabalan.paydar.mohtasham.ezibazi.api_service.Urls;
import sabalan.paydar.mohtasham.ezibazi.model.Device;

public class FireBaseApiService {
  private Context context;
  public FireBaseApiService(Context context){
    this.context = context;
  }




  public void refreshFcmToken(JSONObject object, final onRefreshTokenReceived onRefreshTokenReceived){
    ApiRequest apiRequest = new ApiRequest(context);
    apiRequest.request(ApiRequest.POST, Urls.REFRESH_FCM_TOKEN, object, true, new ApiRequest.onDataReceived() {
      @Override
      public void onReceived(JSONObject response, int status, String message, boolean error) {
        if (error){
          onRefreshTokenReceived.onReceived(status, message, new Device());
          return;
        }

        Device device;
        try {
          status = response.getInt("status");
          message = response.getString("message");
          device = Device.Parser.parse(response.getJSONObject("data"));
          onRefreshTokenReceived.onReceived(status, message, device);
        } catch (JSONException e) {
          e.printStackTrace();
        }
      }
    });
  }







  public interface onRefreshTokenReceived{
    void onReceived(int status, String message, Device device);
  }


}

File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/api_service/firebase/FireBaseApiService.kt
package sabalan.paydar.mohtasham.ezibazi.api_service.firebase

import android.content.Context


import org.json.JSONException
import org.json.JSONObject


import sabalan.paydar.mohtasham.ezibazi.api_service.ApiRequest
import sabalan.paydar.mohtasham.ezibazi.api_service.Urls
import sabalan.paydar.mohtasham.ezibazi.model.Device

class FireBaseApiService(private val context: Context) {


    fun refreshFcmToken(`object`: JSONObject, onRefreshTokenReceived: onRefreshTokenReceived) {
        val apiRequest = ApiRequest(context)
        apiRequest.request(ApiRequest.POST, Urls.REFRESH_FCM_TOKEN, `object`, true, object : ApiRequest.onDataReceived {
            override fun onReceived(response: JSONObject?, status: Int, message: String, error: Boolean) {
                var status = status
                var message = message
                if (error) {
                    onRefreshTokenReceived.onReceived(status, message, Device())
                    return
                }

                val device: Device
                try {
                    status = response!!.getInt("status")
                    message = response.getString("message")
                    device = Device.Parser.parse(response.getJSONObject("data"))
                    onRefreshTokenReceived.onReceived(status, message, device)
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

            }
        })
    }


    interface onRefreshTokenReceived {
        fun onReceived(status: Int, message: String, device: Device)
    }


}
